src/runtime/Object.cpp

```diff
@@ -1542,4 +1542,14 @@ String* Object::optionString(ExecutionState& state)
 
     return new ASCIIString(flags);
 }
+
+bool Object::isRegExp(ExecutionState& state)
+{
+    ObjectGetResult symbol = this->get(state, ObjectPropertyName(state, state.context()->vmInstance()->globalSymbols().match));
+    Value val = symbol.value(state, Value(this));
+    if (!val.isUndefined()) {
+        return val.toBoolean(state);
+    }
```

Comment:
The `5. If argument has a [[RegExpMatcher]] internal slot, return true.` step is missing.